packages/compiler/src/init/scaffold.ts

```diff
@@ -98,9 +98,10 @@ async function writePackageJson(host: CompilerHost, config: ScaffoldingConfig) {
     return;
   }
   const dependencies: Record<string, string> = {};
+  const peerDependencies: Record<string, string> = {};
 
   if (!config.template.skipCompilerPackage) {
-    dependencies["@typespec/compiler"] = "latest";
+    peerDependencies["@typespec/compiler"] = "latest";
```

Comment:
This is not technically a fully correct fix, you only need it to be a peerDependency if you are making a library/emitter. Otherwise a dependency is ok.

Though maybe it doesn't hurt to always do that, our docs do show the difference though
<img width="1064" alt="image" src="https://github.com/microsoft/typespec/assets/1031227/86b8e583-787a-4565-9b49-9beb52d0b781">

<img width="1027" alt="image" src="https://github.com/microsoft/typespec/assets/1031227/24037f13-0d84-4206-9f7c-82cbf863f53a">

Reply:
@timotheeguerin Thanks for the feedback. So, is there a way during generation to find it is a library/emitter? Also, even if it is otherwise, I do not see any issues if it is a peer dependency (instead of dependency)? Is my understanding incorrect? Would it cause any issues if it is a peer dependency?

Reply:
There are actually 3 types of templates:
1. project
2. library
3. emitters

The `empty` and `rest` templates are projects - think of it as a new package that contains the user's TypeSpec files. This should have dependencies on the typespec libraries/compiler.

Library/Emitters should take compilers/typespec libraries as peer dependencies so that the versions the user has defined in their project are used.

It doesn't look like there's an easy way to distinguish which template type you have - maybe a new discriminator field can be added to the templates?

Reply:
Actually - I see now that there are `package.json` files for the [library-ts](https://github.com/microsoft/typespec/blob/main/packages/compiler/templates/library-ts/package.json) and [emitter-ts](https://github.com/microsoft/typespec/blob/main/packages/compiler/templates/emitter-ts/package.json) templates. Does updating the compiler to a peerDependency in those solve this?

Reply:
yeah I think we can just do peerDeps all the time and its ok

Reply:
> yeah I think we can just do peerDeps all the time and its ok

Won't that cause issues if you're not creating an emitter/library? If you're scaffolding a new tsp project, it needs to depend on the compiler directly because you're running the compiler on that project.

For emitters/libraries - they are being imported into a project that would already include the compiler.

Reply:
yeah but peerDepdencies should still be installed for you no?

Reply:
NPM <=3 and >=7 will install them, but some tools like `yarn` won't. It's a mixed bag.

Reply:
What about adding to both peer and dev

Reply:
@sarangan12 what do you think of doing both peer and dev?

Reply:
@timotheeguerin Sure. Adding to both peer and dev makes sense. I will submit the update shortly.

Reply:
@timotheeguerin Updated the code to add the dependency to peer and dev. Thanks

Reply:
the libraries also need to go in both list instead of `depdendencies`